packages/kbn-esql-utils/src/utils/append_to_query.test.ts

```diff
@@ -5,21 +5,116 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
-import { appendToESQLQuery } from './append_to_query';
+import { appendToESQLQuery, appendWhereClauseToESQLQuery } from './append_to_query';
 
-describe('appendToESQLQuery', () => {
-  it('append the text on a new line after the query', () => {
-    expect(appendToESQLQuery('from logstash-* // meow', '| stats var = avg(woof)')).toBe(
-      `from logstash-* // meow
+describe('appendToQuery', () => {
+  describe('appendToESQLQuery', () => {
+    it('append the text on a new line after the query', () => {
+      expect(appendToESQLQuery('from logstash-* // meow', '| stats var = avg(woof)')).toBe(
+        `from logstash-* // meow
 | stats var = avg(woof)`
-    );
-  });
+      );
+    });
 
-  it('append the text on a new line after the query for text with variables', () => {
-    const limit = 10;
-    expect(appendToESQLQuery('from logstash-*', `| limit ${limit}`)).toBe(
-      `from logstash-*
+    it('append the text on a new line after the query for text with variables', () => {
+      const limit = 10;
+      expect(appendToESQLQuery('from logstash-*', `| limit ${limit}`)).toBe(
+        `from logstash-*
 | limit 10`
-    );
+      );
+    });
+  });
+
+  describe('appendWhereClauseToESQLQuery', () => {
+    it('appends a filter in where clause in an existing query', () => {
+      expect(
+        appendWhereClauseToESQLQuery('from logstash-* // meow', 'dest', 'tada!', '+', 'string')
+      ).toBe(
+        `from logstash-* // meow
+| where \`dest\`=="tada!"`
+      );
+    });
+    it('appends a filter out where clause in an existing query', () => {
+      expect(
+        appendWhereClauseToESQLQuery('from logstash-* // meow', 'dest', 'tada!', '-', 'string')
+      ).toBe(
+        `from logstash-* // meow
+| where \`dest\`!="tada!"`
+      );
+    });
+
+    it('appends a where clause in an existing query with casting to string when the type is not string or number', () => {
+      expect(
+        appendWhereClauseToESQLQuery('from logstash-* // meow', 'dest', 'tada!', '-', 'ip')
+      ).toBe(
+        `from logstash-* // meow
+| where \`dest\`::string!="tada!"`
+      );
+    });
+
+    it('appends a where clause in an existing query with casting to string when the type is not given', () => {
+      expect(appendWhereClauseToESQLQuery('from logstash-* // meow', 'dest', 'tada!', '-')).toBe(
+        `from logstash-* // meow
+| where \`dest\`::string!="tada!"`
+      );
+    });
+
+    it('appends a where clause in an existing query checking that the value is not null if the user asks for existence', () => {
+      expect(appendWhereClauseToESQLQuery('from logstash-* // meow', '_exists_', 'dest', '+')).toBe(
```

Comment:
If we are using 

```appendWhereClauseToESQLQuery('from logstash-* // meow', 'dest', 'tada!', '+', 'string')``` 
and 
```appendWhereClauseToESQLQuery('from logstash-* // meow', 'dest', 'tada!', '-', 'string')``` 

for +/- actions, would be great to pass params for "exists" filter as:

```
appendWhereClauseToESQLQuery('from logstash-* // meow', 'dest',  undefined, '_exists_')
```

instead of 

```
appendWhereClauseToESQLQuery('from logstash-* // meow', '_exists_', 'dest', '+')

```

to stay consistent with field name and operation arguments.

Reply:
The weirdness start from Discover to be fair but I see your point 😄 I will adapt.

Reply:
https://github.com/elastic/kibana/pull/181399/commits/a6ddb094a5490fe1f2255d6cb4961e0817ace9f5